net.c

```diff
@@ -293,6 +294,89 @@ void net_if_up(int sockfd, const char *name)
 	}
 }
 
+void net_route_add(int sockfd, const struct route_options *route)
+{
+	struct nlpkt pkt;
+	nlpkt_init(&pkt, sizeof (struct rtmsg));
+
+	pkt.hdr->nlhdr.nlmsg_type = RTM_NEWROUTE;
+	pkt.hdr->nlhdr.nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK | NLM_F_EXCL | NLM_F_CREATE;
+
+	pkt.hdr->rt.rtm_family = route->type;
+	pkt.hdr->rt.rtm_dst_len = route->dst.subnet;
+	pkt.hdr->rt.rtm_src_len = route->src.subnet;
```

Comment:
I'm kind of baffled why a route's src field has a prefix-length at all...  but I guess that's not really in scope for this review, since this is just a field in an existing netlink message... :-/